LWC_Assignment copy/force-app/main/default/lwc/parentPaginator/parentPaginator.js

```diff
@@ -0,0 +1,143 @@
+    @track endingRecord = 0;
+    @track totalRecordCount;
+    @track page = 1;
+    @track totalPage;
+    @track pageSize = 5;
+    @track searchTerm = '';
+    @track editRecordId;
+    @track deleteRecordId;
+
+    @wire(fetchRecords) 
+    wiredFetchRecords(result) {
+        this.handleFetchResult(result);
+    }
+
+    handleRowAction(event) {
+        const actionName = event.detail.action.name;
+        const row = event.detail.row;
+        if (actionName === 'edit') {
+            this.editRecordId = row.Id;
+        } else if (actionName === 'delete') {
+            this.deleteRecordId = row.Id;
+        }
+    }
+
+    handleDeleteClose() {
+        this.deleteRecordId = null;
+        this.searchTerm='';
+    }
+
+    handleUpdateClose() {
+        this.editRecordId = null;
+        this.searchTerm='';
+    }
+
+    handleFetchResult(result) {
+        if (result.data) {
+            this.items = result.data;
+            this.totalRecordCount = result.data.length;
+            this.totalPage = Math.ceil(this.totalRecordCount / this.pageSize);
+            this.updateData();
+        }
+    }
+
+    updateData() {
+        this.endingRecord = this.page * this.pageSize;
+        this.endingRecord = (this.endingRecord > this.totalRecordCount) ? this.totalRecordCount : this.endingRecord;
+        this.data = this.items.slice(this.startingRecord - 1, this.endingRecord);
+    }
+
+    prevHandler(event) {
+        if (this.page > 1) {
+            this.page = this.page - 1;
+            this.startingRecord = (this.page - 1) * this.pageSize + 1;
+            this.updateData();
+        }
+    }
+
+    nextHandler(event) {
+        if (this.page < this.totalPage && this.page != this.totalPage) {
+            this.page = this.page + 1;
+            this.startingRecord = (this.page - 1) * this.pageSize + 1;
+            this.updateData();
+        }
+    }
+
+    handleSearch(event) {
+        this.searchTerm = event.target.value;
+        this.performSearch();
+    }
+
+    performSearch() {
+        if (this.searchTerm) {
+            searchCompanies({ searchTerm: this.searchTerm })
+                .then(result => {
+                    this.items = result;
+                    this.totalRecordCount = result.length;
+                    this.totalPage = Math.ceil(this.totalRecordCount / this.pageSize);
+                    this.updateData();
+                })
+                .catch(error => {
+                    console.error('Error in searching companies:', error);
+                });
+        } else {
+            // Fetch all records when the search term gets empty
+            fetchRecords()
+                .then(result => {
+                    this.items = result;
+                    this.totalRecordCount = result.length;
+                    this.totalPage = Math.ceil(this.totalRecordCount / this.pageSize);
+                    this.updateData();
+                })
+                .catch(error => {
```

Comment:
Let's not use console errors for lwc side, we will use apex debug logs to print errors or toast to show user as well.